include/hip/hcc_detail/hip_runtime.h

```diff
@@ -235,85 +201,15 @@ __device__ int __hip_move_dpp(int src, int dpp_ctrl, int row_mask, int bank_mask
 
 #endif  //__HIP_ARCH_GFX803__ == 1
 
-__host__ __device__ int min(int arg1, int arg2);
-__host__ __device__ int max(int arg1, int arg2);
-
-__device__ void* __get_dynamicgroupbaseptr();
-
-
-/**
- * CUDA 8 device function features
-
- */
-
-
-/**
- * Kernel launching
- */
-
-/**
- *-------------------------------------------------------------------------------------------------
- *-------------------------------------------------------------------------------------------------
- *  @defgroup Fence Fence Functions
- *  @{
- *
- *
- *  @warning The HIP memory fence functions are currently not supported yet.
- *  If any of those threadfence stubs are reached by the application, you should set "export
- *HSA_DISABLE_CACHE=1" to disable L1 and L2 caches.
- *
- *
- *  On AMD platforms, the threadfence* routines are currently empty stubs.
- */
-
-extern __attribute__((const)) __device__ void __hip_hc_threadfence() __asm("__llvm_fence_sc_dev");
-extern __attribute__((const)) __device__ void __hip_hc_threadfence_block() __asm(
-    "__llvm_fence_sc_wg");
-
-
-/**
- * @brief threadfence_block makes writes visible to threads running in same block.
- *
- * @Returns void
- *
- * @param void
- *
- * @warning __threadfence_block is a stub and map to no-op.
- */
-// __device__ void  __threadfence_block(void);
-__device__ static inline void __threadfence_block(void) { return __hip_hc_threadfence_block(); }
-
-/**
- * @brief threadfence makes wirtes visible to other threads running on same GPU.
- *
- * @Returns void
- *
- * @param void
- *
- * @warning __threadfence is a stub and map to no-op, application should set "export
- * HSA_DISABLE_CACHE=1" to disable both L1 and L2 caches.
- */
-// __device__ void  __threadfence(void) __attribute__((deprecated("Provided for compile-time
-// compatibility, not yet functional")));
-__device__ static inline void __threadfence(void) { return __hip_hc_threadfence(); }
-
-/**
- * @brief threadfence_system makes writes to pinned system memory visible on host CPU.
- *
- * @Returns void
- *
- * @param void
- *
- * @warning __threadfence_system is a stub and map to no-op.
- */
-//__device__ void  __threadfence_system(void) __attribute__((deprecated("Provided with workaround
-//configuration, see hip_kernel_language.md for details")));
-__device__ void __threadfence_system(void);
+__device__ inline static int min(int arg1, int arg2) {
```

Comment:
These can't work.  An f32 can not exactly represent every i32.  Just use arg1 < arg2 ? arg1 : arg2.  The compiler will do the rest.